test/pytest/test_bridge.py

```diff
@@ -247,3 +265,103 @@ async def test_dbus_watch(self):
+                                              {'Bridges': ['sudo', 'pseudo'], 'Current': 'none'})
+        null = await self.transport.check_open('null', superuser=True, problem='access-denied')
+        assert null not in self.bridge.open_channels
+
+    async def verify_root_bridge_running(self):
+        await self.transport.assert_bus_props('/superuser', 'cockpit.Superuser',
+                                              {'Bridges': ['sudo', 'pseudo'], 'Current': 'pseudo'})
+        assert self.bridge.superuser_rule.peer is not None
+
+        # try to open dbus on the root bridge
+        root_dbus = await self.transport.check_open('dbus-json3', bus='internal', superuser=True)
+        assert self.bridge.open_channels[root_dbus].name == 'pseudo'
+        assert self.bridge.open_channels[root_dbus].channels == {root_dbus}
+
+        # verify that the bridge thinks that it's the root bridge
+        await self.transport.assert_bus_props('/superuser', 'cockpit.Superuser',
+                                              {'Bridges': ['sudo'], 'Current': 'root'}, bus=root_dbus)
+
+        # close up
+        self.transport.send_close(channel=root_dbus)
+
+    async def test_superuser_dbus(self):
+        await self.start()
+
+        await self.verify_root_bridge_not_running()
+
+        # start the superuser bridge -- no password, so it should work straight away
+        () = await self.transport.check_bus_call('/superuser', 'cockpit.Superuser', 'Start', ['pseudo'])
+
+        await self.verify_root_bridge_running()
+
+        # open a channel on the root bridge
+        root_null = await self.transport.check_open('null', superuser=True)
+
+        # stop the bridge
+        stop = self.transport.send_bus_call(self.transport.internal_bus, '/superuser', 'cockpit.Superuser', 'Stop', [])
+
+        # that should have implicitly closed the open channel
+        await self.transport.assert_msg('', command='close', channel=root_null)
+        assert root_null not in self.bridge.open_channels
+
+        # The Stop method call is done now
+        await self.transport.assert_msg(self.transport.internal_bus, reply=[[]], id=stop)
+
+    async def test_superuser_dbus_pw(self):
+        await self.start()
+        await self.verify_root_bridge_not_running()
+
+        # watch for signals
+        await self.transport.add_bus_match('/superuser', 'cockpit.Superuser')
+        await self.transport.watch_bus('/superuser', 'cockpit.Superuser',
+                                       {'Bridges': ['sudo', 'pseudo'], 'Current': 'none'})
+
+        # start the bridge.  with a password this is more complicated
+        with unittest.mock.patch.dict(os.environ, {"PSEUDO_PASSWORD": "p4ssw0rd"}):
+            start = self.transport.send_bus_call(self.transport.internal_bus, '/superuser', 'cockpit.Superuser', 'Start', ['pseudo'])
+            # first, init state
+            await self.transport.assert_bus_notify('/superuser', 'cockpit.Superuser', {'Current': 'init'})
```

Comment:
I am not sure what the "init" state is actually supposed to mean... I don't think the C bridge cycles through it when calling Start(), for example. I think Current is only "init" during bridge startup, to help reduce some banner flickering during Cockpit login.

Reply:
Confusingly, it's what happens after the bridge has been spawned, but before we see the "init" message.  This is the state we're in when we get authentication requests, for example.

In this state, we have a banner about "Limited privs", but the button is no longer present.

Reply:
Some docs: https://github.com/cockpit-project/cockpit/blob/main/src/bridge/cockpitrouter.c#L1444

Reply:
> In this state, we have a banner about "Limited privs", but the button is no longer present.

Hmm, sounds like a bug.  The whole banner should be hidden when Current == "init".